be/src/exprs/expr.cpp

```diff
@@ -217,12 +223,39 @@ Status Expr::create_expr_tree(ObjectPool* pool, const TExpr& texpr, ExprContext*
     return status;
 }
 
+Status Expr::rewrite_jit_exprs(std::vector<ExprContext*>& expr_ctxs, ObjectPool* pool, RuntimeState* state) {
+    std::vector<ExprContext*> tmp;
+    for (auto ctx : expr_ctxs) {
+        auto* prev_expr = ctx->root();
+        auto* root_expr = &prev_expr;
+        bool replaced = false;
+        auto st = (*root_expr)->replace_compilable_exprs(root_expr, pool, replaced);
+        if (!st.ok()) {
+            LOG(WARNING) << "Can't replace compilable exprs.\n" << st.message() << "\n" << (*root_expr)->debug_string();
+            // Fall back to the non-JIT path.
+            return Status::OK();
+        }
+        auto new_ctx = ctx;
+        if (replaced) {
+            // The node was replaced, so we need to update the context.
+            new_ctx = pool->add(new ExprContext(*root_expr));
```

Comment:
if some exprs are replaced but some one can't be replaced, it will fall back to non-JIT path, should we clear these replaced ExprContext in ObjectPool?

Reply:
I wouldn't take too much memory, and I am not sure whether the old ExprContext is used in other places. We can opt it later once it becomes a issue.